src/network/networkd-json.c

```diff
@@ -1044,6 +1045,31 @@ static int dhcp_server_append_json(Link *link, JsonVariant **v) {
         return json_append_one(v, "DHCPServer", w);
 }
 
+static int dhcp6_client_vendor_options_append_json(Link *link, JsonVariant **v) {
+        _cleanup_(json_variant_unrefp) JsonVariant *array = NULL;
+        int r;
+
+        assert(link);
+        assert(v);
+
+        if (!link->dhcp6_lease)
+                return 0;
+
+        LIST_FOREACH(options, option, link->dhcp6_lease->vendor_options) {
```

Comment:
Rather than reading internal element of `sd_dhcp6_lease`, please introduce a simple getter `sd_dhcp6_lease_get_vendor_options()`.

Reply:
I tried to make it simpler but traversing and append the data to the JSON. I felt having sd_dhcp6_lease_get_vendor_options() will be redundant here. Kindly let me know if you think otherwise.

Reply:
I do not think it is redundant. When the getter is called first time, we should sort the options (and cache the result). When called again, the getter can simply pass the previously sorted result.

Reply:
Fixed it.